data-mongodb/src/main/java/io/micronaut/data/mongodb/init/MongoReactiveCollectionsCreator.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.mongodb.init;
+
+import com.mongodb.reactivestreams.client.MongoDatabase;
+import io.micronaut.configuration.mongo.core.AbstractMongoConfiguration;
+import io.micronaut.context.BeanLocator;
+import io.micronaut.context.annotation.Context;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.data.model.PersistentEntity;
+import io.micronaut.data.model.runtime.RuntimeEntityRegistry;
+import io.micronaut.data.mongodb.conf.MongoDataConfiguration;
+import io.micronaut.data.mongodb.conf.RequiresReactiveMongo;
+import io.micronaut.data.mongodb.database.ReactiveMongoDatabaseFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * MongoDB's reactive collections creator.
+ *
+ * @author Denis Stepanov
+ * @since 3.3
+ */
+@Context
+@Internal
+@RequiresReactiveMongo
+@Requires(property = MongoDataConfiguration.CREATE_COLLECTIONS_PROPERTY, value = StringUtils.TRUE)
+public final class MongoReactiveCollectionsCreator extends AbstractMongoCollectionsCreator<MongoDatabase> {
```

Comment:
Doesn;t it make more sense to use a `BeanCreatedEventListener<MongoClient>` rather than rely on `@Context` and `@PostConstruct`?

Reply:
The problem here is that we need to use `MongoDatabaseFactory` and it's also using `MongoClient` so it will not load.